sourcetable.go

```diff
@@ -123,3 +127,163 @@ func (m MountEntry) String() string {
 		m.CountryCode, m.Latitude, m.Longitude, nmea, solution, m.Generator, m.Compression,
 		m.Authentication, fee, m.Bitrate, m.Misc)
 }
+
+// DecodeSourcetable parses a sourcetable from an ioreader into a ntrip style source table.
+func DecodeSourcetable(in []byte) (Sourcetable, []error) {
```

Comment:
I think "Decode" might not be technically correct. Either Parse or Unmarshal?

Parse is probably better given we use the method "String" to convert the struct to the representation which you're parsing here (rather than Marshal, which would feel weird here I think).